xarray/tests/test_backends.py

```diff
@@ -278,6 +278,32 @@ def test_roundtrip_datetime_data(self):
             self.assertEquals(actual.t0.encoding['units'],
                               'days since 1950-01-01')
 
+    def test_roundtrip_netcdftime_datetime_data(self):
+        from .test_coding import _non_standard_netcdftime_date_types
```

Comment:
I think its worth opening an issue with `netcdftime` and we can talk with @jswhit to see if using a zero padded year repr makes sense.